controllers/storageclass/storageclass_controller_test.go

```diff
@@ -216,6 +221,86 @@ var _ = Describe("StorageClassWatcher", func() {
 			}, timeout, interval).Should(BeTrue())
 		})
 
+		It("should get FlashSystemCluster by StorageClass successfully", func() {
+			ctx := context.TODO()
+
+			By("creating a new secret with topology awareness")
+			secret := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      topologySecretName,
+					Namespace: namespace,
+				},
+				Data: map[string][]byte{
+					"config": []byte(topologySecretConfigData),
+				},
+			}
+			Expect(k8sClient.Create(ctx, secret)).Should(Succeed())
+
+			By("By creating a new topology StorageClass")
+			topologySc := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      topologyStorageClassName,
+					Namespace: namespace,
+				},
+				Provisioner: util.CsiIBMBlockDriver,
+				Parameters: map[string]string{
+					"by_management_id":                    byManagementIdData,
+					"SpaceEfficiency":                     spaceEff,
+					"pool":                                poolName,
+					"csi.storage.k8s.io/secret-name":      topologySecretName,
+					"csi.storage.k8s.io/secret-namespace": namespace,
+					"csi.storage.k8s.io/fstype":           fsType,
+					"volume_name_prefix":                  volPrefix,
+				},
+			}
+			Expect(k8sClient.Create(ctx, topologySc)).Should(Succeed())
+
+			By("By querying the created topology StorageClass")
+			topologyScLookupKey := types.NamespacedName{
+				Name:      topologyStorageClassName,
+				Namespace: namespace,
+			}
+			createdTopologySc := &storagev1.StorageClass{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, topologyScLookupKey, createdTopologySc)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("By querying the ConfigMap to verify the topology StorageClass is added")
+			cmLookupKey := types.NamespacedName{
+				Name:      util.PoolConfigmapName,
+				Namespace: namespace,
+			}
+			createdCm := &corev1.ConfigMap{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cmLookupKey, createdCm)
+				if err == nil {
+					var sp util.FlashSystemClusterMapContent
+					err = json.Unmarshal([]byte(createdCm.Data[FlashSystemName]), &sp)
+					if err == nil {
+						return sp.ScPoolMap[topologyStorageClassName] == topologyPoolName
+					}
+				}
+				return false
+			}, timeout, interval).Should(BeTrue())
+
+			Expect(k8sClient.Delete(ctx, createdTopologySc)).Should(Succeed())
+
+			By("By querying the ConfigMap to verify the topology StorageClass is deleted")
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, cmLookupKey, createdCm)
+				if err == nil {
+					var sp util.FlashSystemClusterMapContent
+					err = json.Unmarshal([]byte(createdCm.Data[FlashSystemName]), &sp)
+					if err == nil {
+						return sp.ScPoolMap[topologyStorageClassName] == ""
```

Comment:
I think the verification here should be like we did for regular StorageClass (check if key exist)